recipes-bsp/u-boot/files/0010-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,223 @@
++	char config_3;
++}m2_bkey_config_t;
++
++
++typedef struct serdes_select_mux {
++	char sel_usb30_pcie0_lane0;
++	char sel_pcie1_pcie0;
++	char sel_usb30_pcie0_lane1;
++}serdes_select_mux_t;
++
++typedef struct key_config {
++	m2_bkey_config_t m2_bkey_config;
++	serdes_select_mux_t m2_pinmux_select;
++	serdes_select_t serdes_select;
++}__packed key_config_t;
++
++static key_config_t config_table[] = {
++	{{0,1,0,0}, {0,0,1}, {PCIE0_LANE0, PCIE0_LANE1}},
++	{{0,0,1,0}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{{0,1,1,0}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{{1,0,0,1}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{{1,1,0,1}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{{0,0,0,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{{0,1,0,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{{0,0,1,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{{0,1,1,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{{1,0,1,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++};
++
++static gpioConfig_t m2_pinmux_config[PCIE_MUX_CONTROL] = {
++	{"SEL_USB30_PCIE0_LANE0", "gpio@600000_88"},
++	{"SEL_PCIE1_PCIE0",       "gpio@600000_82"},
++	{"SEL_USB30_PCIE0_LANE1", "gpio@600000_89"},
++};
++
++static gpioConfig_t m2_key_config[M2_CONFIG_LEN] = {
++	{"KEY_CONFIG_0", "gpio@601000_18"},
++	{"KEY_CONFIG_1", "gpio@601000_19"},
++	{"KEY_CONFIG_2", "gpio@601000_88"},
++	{"KEY_CONFIG_3", "gpio@601000_89"},
++};
++
++static gpioConfig_t m2_enable_power = {"P3V3_M2_EN", "gpio@601000_17"};
++
++static int get_pinvalue(const char *gpio_name,const char *label)
++{
++	struct gpio_desc gpio;
++
++	if (dm_gpio_lookup_name(gpio_name, &gpio) < 0 ||
++	dm_gpio_request(&gpio, label) < 0 ||
++	dm_gpio_set_dir_flags(&gpio, GPIOD_IS_IN) < 0)
++		return false;
++
++	return dm_gpio_get_value(&gpio);
++}
++
++static void set_pinvalue(const char *gpio_name,const char *label,int value)
++{
++	struct gpio_desc gpio;
++
++	if (dm_gpio_lookup_name(gpio_name, &gpio) < 0 ||
++	dm_gpio_request(&gpio, label) < 0 ||
++	dm_gpio_set_dir_flags(&gpio, GPIOD_IS_OUT) < 0) {
++		pr_err("IOT2050:Can not set the pin value\n");
++		return ;
++	}
++	dm_gpio_set_value(&gpio,value);
++}
++
++static m2_bkey_config_t get_m2_config(void)
++{
++	m2_bkey_config_t result;
++
++	result.config_0 = get_pinvalue(m2_key_config[0].gpio_name, m2_key_config[0].label);
++	result.config_1 = get_pinvalue(m2_key_config[1].gpio_name, m2_key_config[1].label);
++	result.config_2 = get_pinvalue(m2_key_config[2].gpio_name, m2_key_config[2].label);
++	result.config_3 = get_pinvalue(m2_key_config[3].gpio_name, m2_key_config[3].label);
++
++	return result;
++}
++
++static serdes_select_t get_key_config_set_m2_pinmux(void)
++{
++	int i;
++	serdes_select_t serdes_select = {USB30, PCIE1_LANE0}; /* default setting */
++	int config_table_size= sizeof(config_table)/sizeof(config_table[0]);
++	m2_bkey_config_t config = get_m2_config();
++
++	printf("B-KEY:");
++	printf("%d",config.config_0);
++	printf("%d",config.config_1);
++	printf("%d",config.config_2);
++	printf("%d",config.config_3);
++	printf("\n");
```

Comment:
Debug left-overs?

Reply:
> Debug left-overs?

yes,  but It would be better to use the debug macro. would change later